analyses/cell-type-ewings/exploratory_analysis/annotation_notebooks/SCPCS000492/SCPCL000824_tumor-gene-expression-exploration.Rmd

```diff
@@ -0,0 +1,749 @@
+```
+
+For each gene set, several possible thresholds are calculated. 
+The dotted lines over the histogram indicate the distribution and the corresponding vertical dotted line (same color) indicates the possible threshold. 
+The thicker line indicates the chosen threshold by the algorithm. 
+This corresponds to the highest value that reduces the false positives. 
+
+Here we can see a bimodal distribution of all gene sets except `SILIGAN`, which is to be expected based on the original distribution plots. 
+
+For the remaining analysis we will use the results from running `AUCell` with only the marker gene list. 
+Let's compare the assignments from our validated classifications (determined by taking the consensus between `InferCNV`, and `CopyKAT` for `SCPCL000822`) to using `AUCell` with `SCPCL000822`. 
+
+```{r}
+# compare AUCell classification vs cnv classification in ref
+# pull out tumor cells found by AUCell
+tumor_cells <- ref_assignment$markers$assignment
+
+ref_classification_df <- ref_classification_df |> 
+  # add new column with auc classification 
+  dplyr::mutate(auc_classification = dplyr::if_else(barcodes %in% tumor_cells, "Tumor", "Normal"))
+
+# filter out any ambiguous cells 
+caret_df <- ref_classification_df |> 
+  dplyr::filter(tumor_cell_classification != "Ambiguous")
+
+# compare using a confusion matrix
+caret::confusionMatrix(
+    table(
+      caret_df$tumor_cell_classification, 
+      caret_df$auc_classification)
+  ) 
+```
+
+It looks like we get pretty good agreement between our original classification and the updated `AUCell` classification using the marker gene list.
+
+Now we repeat on `SCPCL000824`. 
+
+```{r}
+# get new auc and plot thresholds 
+new_auc <- AUCell::AUCell_run(counts(sce), collection)
+new_assignment <- AUCell::AUCell_exploreThresholds(new_auc, assign = TRUE)
+```
+
+Here we do not see any bimodal distributions and that most cells are considered to not be expressing the specified gene sets, which doesn't fit with our previous findings. 
+I expect that without a bimodal distribution this model does not do a good job of calculating a threshold. 
+
+```{r}
+ref_auc_df <- ref_auc@assays@data$AUC |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("gene_set") |> 
+  tidyr::pivot_longer(!gene_set, 
+                      names_to = "barcodes", 
+                      values_to = "auc") |> 
+  #dplyr::filter(gene_set == "markers") |> 
+  dplyr::mutate(sample = "SCPCL000822")
+
+new_auc_df <- new_auc@assays@data$AUC |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("gene_set") |> 
+  tidyr::pivot_longer(!gene_set, 
+                      names_to = "barcodes", 
+                      values_to = "auc") |> 
+  #dplyr::filter(gene_set == "markers") |> 
+  dplyr::mutate(sample = "SCPCL000824")
+
+all_auc_df <- dplyr::bind_rows(list(ref_auc_df, new_auc_df))
+
+ggplot(all_auc_df, aes(x = auc, color = sample)) +
+  geom_density() +
+  facet_grid(rows = vars(gene_set))
```

Comment:
Hopefully making the distributions easier to see
```suggestion
  facet_grid(rows = vars(gene_list),
             scales = "free_y")
```

Reply:
I'm not entirely sure what you mean by this comment since the `gene_set` column exists in the data frame but `gene_list` does not? Each row already is a gene set?

Reply:
A bad copy paste! I just wanted the `free_y`